budget_control_operating_unit/models/base_budget_move.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class BaseBudgetMove(models.AbstractModel):
+    _inherit = "base.budget.move"
+
+    operating_unit_id = fields.Many2one(
+        comodel_name="operating.unit",
+        compute="_compute_operating_unit_id",
+        store=True,
+        index=True,
+    )
+
+    @api.depends("analytic_account_id.operating_unit_ids")
+    def _compute_operating_unit_id(self):
+        for rec in self:
+            if len(rec.analytic_account_id.operating_unit_ids) > 1:
+                raise UserError(_("Analytic can't have many operating unit."))
+            rec.operating_unit_id = (
```

Comment:
Please don't use OU form analytic account. OU should come from document line that is creating this budget move.

What if in the future, we want to create budget move without analytic account (it is possible too). If you want to use related, it should related with document line (store = True, so it won't change).

Reply:
Giving priority to detailed lines and not the header is concept of ERP. That's why they try to push everything down to the lines.
https://github.com/OCA/operating-unit/blob/12.0/purchase_request_operating_unit/model/purchase_request.py#L64